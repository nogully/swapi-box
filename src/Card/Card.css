.Card {
  background-color: #0A0847;
  border: 1px solid #0A0847;
  height: 350px;
  margin: 10px;
  opacity: .8;
  padding: 0px 10px;
  width: 250px;
}

.active {
  background-color: purple;
}

.Card h1 {
  color: #feda4a;
}