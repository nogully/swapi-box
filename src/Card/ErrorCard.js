import React from 'react';

const ErrorCard = ({ data, favoriteCard, isActive, key }) => {
  const attributes = Object.keys(data).filter( attribute => attribute !== 'name')
  const elements = attributes.map( (attribute, index) => {
    return <p key={index}>{`${attribute} : ${data[attribute]}`}</p>
  })

  return (
    <article className={ "Error " } >
      <h1>Error</h1>
      <p>There are no favorites</p>
    </article>
  )

}

export default ErrorCard;
