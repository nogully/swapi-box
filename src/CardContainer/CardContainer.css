.CardContainer {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  max-width: 1000px;
  min-width: 1000px;
}