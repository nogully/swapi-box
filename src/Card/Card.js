import React from 'react';
import Button from '../Button/Button'
import './Card.css'

const Card = (props) => {

  return (
    <article className={"Card " + props.active} >
      <h1>{props.name}</h1>
      <h3>{props.species}</h3>
      <h3>Homeworld: {props.homeworld}</h3>
      <h3>Population: {props.population}</h3>
      <Button buttonType='light'
              buttonText='Favorite'
              function={props.favoriteCard}
              name={props.name} />
    </article>
  )

}

export default Card;