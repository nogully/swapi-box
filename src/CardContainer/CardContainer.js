import React from 'react';
import Card from '../Card/Card'
import ErrorCard from '../Card/ErrorCard'
import './CardContainer.css'
import PropTypes from 'prop-types'

const CardContainer = ({ category, cardArray, favorites, favoriteCard}) => { 
  const cards = cardArray.map( card => {
    const isActive = favorites.includes(card) ? 'active' : '';
    return (
      <Card data={ card }
            favoriteCard={favoriteCard}
            isActive={isActive}
            key={ card.name } />
    ) 
  })          

  return (
    <div className="CardContainer">
      { category === 'favorites' && !favorites.length ? <ErrorCard /> : null }
      { cards }
    </div>
  )
}

CardContainer.propTypes = {
  category: PropTypes.string.isRequired,
  cardArray: PropTypes.array.isRequired,
  favoriteCard: PropTypes.func.isRequired, 
  favorites: PropTypes.array.isRequired
}

export default CardContainer;