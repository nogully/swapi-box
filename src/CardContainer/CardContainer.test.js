 /* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import CardContainer from './CardContainer';
import Card from '../Card/Card';
import ErrorCard from '../Card/ErrorCard';
import { shallow, mount } from 'enzyme';

describe('CardContainer', () => {
  let wrapper
  let mockArray
  let mockFaves
  let favoriteCard

  beforeEach(() => {  
    mockArray = [ { "name":'Endor', 
                    "terrain": 'forest', 
                    "climate": 'temperate', 
                    "population": '30000000',
                    "residents": 'Ewoks' } ]
    mockFaves = [ { "name":'Earth', 
                    "terrain": 'forest', 
                    "climate": 'temperate', 
                    "population": '700000000',
                    "residents": 'Humans' } ]
    favoriteCard = jest.fn()
    wrapper = shallow(<CardContainer category="favorites"
                                     favoriteCard={ favoriteCard } 
                                     favorites={ mockFaves } 
                                     cardArray={ mockArray }/>);
  });

  it('exists and matches snapshot', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('shows favorite cards if favorites array has items', () => {
    expect(wrapper.find(Card).length).toEqual(1)  
  }) 

  it('shows an ErrorCard if category is favorites and favorites array is empty', () => {
    let emptyFaves = []
    wrapper = shallow(<CardContainer category="favorites"
                                     favoriteCard={ favoriteCard } 
                                     favorites={ emptyFaves } 
                                     cardArray={ mockArray }/>);
    expect(wrapper.find(ErrorCard).length).toEqual(1)  
  })
})