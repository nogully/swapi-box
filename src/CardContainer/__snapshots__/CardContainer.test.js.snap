// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer exists and matches snapshot 1`] = `
<div
  className="CardContainer"
>
  <Card
    data={
      Object {
        "climate": "temperate",
        "name": "Endor",
        "population": "30000000",
        "residents": "Ewoks",
        "terrain": "forest",
      }
    }
    favoriteCard={[Function]}
    isActive=""
    key="Endor"
  />
</div>
`;
